pkg/miniogw/gateway-storj.go

```diff
@@ -249,7 +253,31 @@ func (s *storjObjects) MakeBucketWithLocation(ctx context.Context,
 func (s *storjObjects) PutObject(ctx context.Context, bucket, object string,
 	data *hash.Reader, metadata map[string]string) (objInfo minio.ObjectInfo,
 	err error) {
+	log.Printf("Inside PutObject")
+
 	defer mon.Task()(&ctx)(&err)
+	ctx, cancel := context.WithCancel(ctx)
+
+	/* create a signal of type os.Signal */
+	c := make(chan os.Signal, 0x01)
+
+	/* register for the os signals */
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+
+	defer func() {
+		signal.Stop(c)
+		cancel()
+	}()
+
+	go func() (minio.ObjectInfo, error) {
+		select {
+		case <-c:
+			log.Printf("start the cleanup")
```

Comment:
If you keep this log statement, I think it should just say something like "Cleaning up..."
But I'm not sure whether we want to be printing out anything for this.